tests/functional/i/iterating_modified.py

```diff
@@ -0,0 +1,28 @@
+item_list = [1, 2, 3]
+for item in item_list:
+    item_list.append(item)  # [iterating-modified-list]
```

Comment:
Could you also add tests for `iterating-modified-set` and `dict`?

Reply:
once they are implemented obviously I will